[jdbc] Service fails in activation when having persisted items with invalid name

Reported in the forum:
https://community.openhab.org/t/openhab-3-4-release-discussion/142257/115

## Expected Behavior
The persistence service should work, but skip and warn about invalid item names.

## Current Behavior
Exception is thrown during `@Activate`:

```
2022-12-23 11:40:40.096 [ERROR] [jdbc.internal.JdbcPersistenceService] - bundle org.openhab.persistence.jdbc:3.4.0 (304)[org.openhab.persistence.jdbc.internal.JdbcPersistenceService(382)] : The activate method has thrown an exception
java.lang.IllegalArgumentException: 12 is not a valid item name
```

## Possible Solution
Skip invalid items.

## Steps to Reproduce
For configurations not using real names, manipulate an item in the items manage table by appending e.g. a "%" to its name:
![image](https://user-images.githubusercontent.com/19519842/209413774-90c0e6a3-853c-406b-a521-f37e4d4ccd3d.png)

Then start the service.

--- a/bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/internal/JdbcMapper.java
+++ b/bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/internal/JdbcMapper.java
@@ -27,6 +27,7 @@ import org.eclipse.jdt.annotation.Nullable;
 import org.knowm.yank.Yank;
 import org.openhab.core.i18n.TimeZoneProvider;
 import org.openhab.core.items.Item;
+import org.openhab.core.items.ItemUtil;
 import org.openhab.core.persistence.FilterCriteria;
 import org.openhab.core.persistence.HistoricItem;
 import org.openhab.core.persistence.PersistenceItemInfo;
@@ -340,6 +341,11 @@ public class JdbcMapper {
             }
         } else {
             for (ItemsVO vo : getItemIDTableNames()) {
+                String itemName = vo.getItemName();
+                if (!ItemUtil.isValidItemName(itemName)) {
+                    logger.warn("Skipping invalid item name {}", itemName);
+                    continue;
+                }
                 itemNameToTableNameMap.put(vo.getItemName(),
                         namingStrategy.getTableName(vo.getItemId(), vo.getItemName()));
             }
@@ -429,6 +435,10 @@ public class JdbcMapper {
             }
             oldNewTableNames = new ArrayList<>();
             for (String itemName : itemTables) {
+                if (!ItemUtil.isValidItemName(itemName)) {
+                    logger.warn("JDBC::formatTableNames: Skipping invalid item name {}", itemName);
+                    continue;
+                }
                 ItemsVO isvo = new ItemsVO();
                 isvo.setItemName(itemName);
                 isvo.setItemsManageTable(conf.getItemsManageTable());

